sql/rpl_filter.cc

```diff
@@ -749,6 +836,36 @@ Rpl_filter::rewrite_db_is_empty()
 }
 
 
+I_List<i_string_pair>*
+Rpl_filter::get_rewrite_db()
+{
+  return &rewrite_db;
+}
+
+
+void
+Rpl_filter::db_rewrite_rule_ent_list_to_str(String* str, I_List<i_string_pair>* list)
+{
+  I_List_iterator<i_string_pair> it(*list);
+  i_string_pair* s;
+
+  str->length(0);
+
+  while ((s= it++))
```

Comment:
Two things in this loop to consider:
 1. The separate appends for `-` and `>` can be combined into `append(STRING_WITH_LENGTH("->"))`
 2. Always adding the delimiter can lead to an unnecessary reallocation of the memory to hold the string. If you are avoiding the branch check to see if the string is empty before adding the delimiter, you can initialize a delimiter string and size to "," and 0 for the first iteration, and then set the length to '1' for all other iterations to be both memory and CPU optimized. 
 
 The result would look something like..
```
const char *delimiter= ",";
size_t delim_len= 0;
while ((s= it++))
{
  str->append(delimiter, delim_len); // Will exit early with no change if delim_len is 0
  str->append(s->key, strlen(s->key));
  str->append(STRING_WITH_LENGTH("->"));
  str->append(s->val, strlen(s->val));
  delim_len= 1;
}

// No need for chop()
```